sympy/polys/rootoftools.py

```diff
@@ -970,8 +970,8 @@ def _eval_Eq(self, other):
         # is_real value of the CRootOf instance.
         if type(self) == type(other):
             return sympify(self == other)
-        if not (other.is_number and not other.has(AppliedUndef)):
-            return S.false
+        if not other.is_number:
+            return None
```

Comment:
Change this to `return None if other.is_constant else False`

Reply:
Actually sorry I see that this might work okay. Have you tested it on your computer? Is it better than the other suggestions?

Reply:
Have you tested it on your computer? Is it better than the other suggestions?

I have tested it and I think it is a better suggestion.

Reply:
Those were not really yes/no questions. Can you explain what the difference is? Why is this version better? What examples are handled differently?

Reply:
`return None` is a fast, conservative (safe) result; using `is_constant()` method by default will inolve simplification but `simplify=False` can be passed. In that case, two random numerical evaluations will be attempted to see if the results are the different (hence not constant) or not (return None since it could be a fluke). We could think of Eq as a "low hanging fruit" method which tests for really obvious cases and `equals/simplification/expression manipulation` as being the much more expensive 'help me out, here' methods.

I am happy with this as it is. It would be interesting to see if `foo.is_constant(simplify=False)` can be used with any profit here or in other `+eval_Eq` methods.

Reply:
I'm also happy with this as it is. The current version makes sense to me even if further optimisations might be possible.